src/buildstream/_sourcecache.py

```diff
@@ -254,5 +256,7 @@ def _push_source(self, source_ref, remote):
 
         except grpc.RpcError as e:
             if e.code() != grpc.StatusCode.RESOURCE_EXHAUSTED:
-                raise SourceCacheError("Failed to push source with status {}: {}".format(e.code().name, e.details()))
+                raise SourceCacheError(
```

Comment:
This function `_push_source()` needs the same treatment as `_pull_source()` above:

* It is impossible to receive `grpc.RpcError` here, as this is already handled in `AssetCache.push_directory()`
* We need to handle `AssetCacheError` instead, and raise the temporary `SourceCacheError`
*